server/src/main/java/org/opensearch/snapshots/SnapshotShardsService.java

```diff
@@ -292,16 +292,14 @@ private void startNewShards(SnapshotsInProgress.Entry entry, Map<ShardId, IndexS
                             public void onResponse(String newGeneration) {
                                 assert newGeneration != null;
                                 assert newGeneration.equals(snapshotStatus.generation());
-                                if (logger.isDebugEnabled()) {
-                                    final IndexShardSnapshotStatus.Copy lastSnapshotStatus = snapshotStatus.asCopy();
-                                    logger.debug(
-                                        "snapshot [{}] completed to [{}] with [{}] at generation [{}]",
-                                        snapshot,
-                                        snapshot.getRepository(),
-                                        lastSnapshotStatus,
-                                        snapshotStatus.generation()
+                                final IndexShardSnapshotStatus.Copy lastSnapshotStatus = snapshotStatus.asCopy();
```

Comment:
This change moves this new object creation outside of the isDebug check. Is there a way to move it down into the lambda where it's used?